### PR TITLE
Add oEmbed endpoint to charted

### PR DESCRIPTION
**Pull request by screeley**

### CODE REVIEW

File: src/public/embed.js [hunk 1]
```diff
@@ -40,19 +40,29 @@ var CHARTED
...
-      link.setAttribute('href', iframe.getAttribute('src'))
 
-      if (ev.origin !== link.origin) {
-        return
+      // Use the standardized version of postMessage iframe.resize.
+      var data
+      try {
+        data = JSON.parse(ev.data)
+      } catch (e) {
+        return false
```
Comment (valueof) [inline src/public/embed.js]: Do you need to return false here? I think just returning will suffice.

File: src/shared/utils.js [hunk 2]
```diff
@@ -17,6 +18,17 @@ function getChartId(params: t_CHART_PARAM): string {
   return sha1(JSON.stringify(params), /* short */ true)
 }
 
+/*
+* Returns the Chart ID from a given URL.
+*/
+function parseChartId(url:string): string {
+  const match = (/\/(?:c|embed)\/(\w+)(?:|\?.*)?/).exec(url)
+  if (match){
```
Comment (valueof) [inline src/shared/utils.js]: Can you add a space before the curly brace, please?

Comment (valueof) [inline src/shared/utils.js]: Also, will `return match ? match[1] : null` work here?

**Comment by valueof** [comment]
Thank you! Left a couple of small questions.

**Comment by screeley** [comment]
Updated, thanks @valueof!

**Comment by valueof** [comment]
LG, thanks!

**Pull request merged**